applications/fishing-map/src/features/resources/resources.slice.ts

```diff
@@ -0,0 +1,69 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit'
+import { AsyncReducerStatus } from 'types'
+import { RootState } from 'store'
+import { trackValueArrayToSegments, Field } from '@globalfishingwatch/data-transforms'
+import GFWAPI from '@globalfishingwatch/api-client'
+import { DataviewDatasetConfig, DatasetTypes } from '@globalfishingwatch/dataviews-client'
+
+export interface ResourceQuery {
+  url: string
+  dataviewId: number
+  datasetConfig: DataviewDatasetConfig
+  datasetType: DatasetTypes
+}
+
+export interface Resource extends ResourceQuery {
+  data: unknown
+}
+
+interface ResourcesState {
+  status: AsyncReducerStatus
+  resources: Record<string, Resource>
+}
+
+const initialState: ResourcesState = {
+  status: AsyncReducerStatus.Idle,
```

Comment:
I'd move this to the internal resource property as it is requested [in batch](https://github.com/GlobalFishingWatch/frontend/pull/109/files#diff-283a0939f4dbf6c8a3d9bf4ce071add1R53) will be nice to have the status of each request